common/src/main/kotlin/fr/agrouagrou/common/extensions/Player.kt

```diff
@@ -0,0 +1,23 @@
+package fr.agrouagrou.common.extensions
+
+import fr.agrouagrou.common.player.Player
+import fr.agrouagrou.common.player.Role
+import fr.agrouagrou.proto.PlayerReply
+import fr.agrouagrou.proto.PlayerRole
+import fr.agrouagrou.proto.playerReply
+
+fun Player.toProto(): PlayerReply =
+    playerReply {
+        id = this@toProto.id.toString()
+        username = this@toProto.username
+        alive = this@toProto.alive
+        role = this@toProto.role.toProto()
+    }
+
+fun Role.toProto(): PlayerRole =
+    when (this) {
+        Role.Villager -> PlayerRole.VILLAGER
+        Role.Werewolf -> PlayerRole.WEREWOLF
+        Role.FortuneTeller -> PlayerRole.FORTUNE_TELLER
+        is Role.Witch -> PlayerRole.WITCH
```

Comment:
is there a reason as to why only `Role.Witch` is using the `is` operator?

Reply:
Yup, if you look at the definition [of the roles](https://github.com/mmoreiradj/agrouagrou/blob/b45ec7691367fa0c23857124e963c9f935126a6f/common/src/main/kotlin/fr/agrouagrou/common/player/Role.kt), you can see that the Witch is a class (because it holds the amount of potions that a player has) whereas the other roles are objects (because they are stateless).

In the case of the Witch, we need to check if the role of the player `is` an instance of the `Role.Witch` class, whereas we can simply check for the other roles if they are equal to the (single) instance of that role.